free-2.md

```diff
@@ -11,41 +11,100 @@ _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help 
 them?_
 
+The process of writing a program to solve a recursive solution efficiently
+(using a Dynamic Programming (DP) solution) requires a lot of boilerplate code
+and opportunity for simple errors to be introduced.  The individuals using
+this language would be anyone who hopes to implement a solution to a recursive
+algorithm more efficiently.
+
+Hopefully their experience would be transformed from thinking about how
```

Comment:
I recall that a hackathon from a few years ago used AST hacking in Python to convert most arbitrary recursive functions to dynamic programming variants